packages/block-editor/README.md

```diff
@@ -12,6 +12,58 @@ npm install @wordpress/block-editor --save
 
 _This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for ES2015+ such as lower versions of IE then using [core-js](https://github.com/zloirock/core-js) or [@babel/polyfill](https://babeljs.io/docs/en/next/babel-polyfill) will add support for these methods. Learn more about it in [Babel docs](https://babeljs.io/docs/en/next/caveats)._
 
+## Usage
+
+```js
+import {
+	BlockEditorProvider
+	BlockList
+} from '@wordpress/block-editor';
+import { useState } from '@wordpress/element';
+
+function MyEditorComponent () {
+	const [ blocks, updateBlocks ] = useState( [] );
+
+	return (
+		<BlockEditorProvider
+			value={ blocks }
+			onInput={ updateBlocks }
+			onChange={ updateBlocks }
```

Comment:
For someone who doesn't care about "Undo" levels, I wonder if the simplest example can just include `onInput` (or conversely, for one who cares only about important updates, only include `onChange`).

Though as I write this, I guess it assumes that `onInput` and `onChange` aren't mutually exclusive (i.e. `onInput` would always precede an `onChange`). I think this was an original intention, but by the final implementation it didn't end up this way:

https://github.com/WordPress/gutenberg/blob/d9768ad4294ffc08bed68459d342fa746f63aea4/packages/block-editor/src/components/provider/index.js#L95-L99

The original thinking being that you could have `onInput` and `onChange`, _or_ just individually one or the other of `onInput` or `onChange`.

Reply:
Good thoughts, and yeah I think you're right, it would have been better to have the two calls at the same time. I recall something related to the state making this harder for us but I think we can probably revisit specifically.

Reply:
IIRC, #14105 was partly toward this goal as well.